### ISSUE TITLE
[Download files] Download files from an AWS S3 bucket.

### ISSUE THREAD

**Issue by LoicPoullain** [FoalTS/foal#421]
# Issue

We currently have a solution to download a file from the file system (`createHttpResponseFile`). But we do not have a way to download in streaming a file from an AWS S3 bucket.

# Possible solution

Create a package `@foal/aws` and a `createHttpResponseFileFromS3` function.

```typescript
import { createHttpResponseFileFromS3 } from '@foal/aws';
import { Context, Get } from '@foal/core';

class MyController {

  @Get('/my-pdf')
  getMyPdf(ctx: Context) {

    return createHttpResponseFileFromS3({
      bucket: 'name_of_the_bucket',
      key: 'name_of_the_file',
      // Tell the browser to download the pdf and not to display it.
      forceDownload: true
      // Default name used by the browser when saving the file to the client disk.
      filename: 'my_pdf.pdf',
    });

  }

}
```

**Comment by fer22f** [comment]
I'm planning on using DigitalOcean's Spaces and it has AWS S3 bucket compatible API, so I believe it shouldn't be too hard to also support that (no idea if it would fit `@foal/aws` however).

**Comment by LoicPoullain** [comment]
PR merged. It will be released with version 1.6.

@fer22f the new feature should support both S3 and DigitalOcean's Spaces.
